src/Microsoft.DotNet.Build.Tasks/Targets/packages.targets

```diff
@@ -76,4 +80,47 @@
       <_TempPackageVersion />
     </PropertyGroup>
   </Target>
+
+  <Import Condition="Exists('$(LocalPublishPropsFile)')"  Project="$(LocalPublishPropsFile)"/>
+  
+  <!-- Publish the packages locally -->
+  <Target Name="LocalPackagePublish" Condition="'@(Packages)'!=''"
+          Inputs="%(Packages.Identity)\%(Packages.PackageName).%(Packages.PackageVersion).nupkg;%(Packages.InstallLocation)\%(Packages.PackageName).%(Packages.PackageVersion)" 
+          Outputs="%(Packages.InstallLocation)\localpublish.%(Packages.PackageName).%(Packages.PackageVersion).txt" >
+    
+    <!-- Generate the packages.config -->
+    <ItemGroup>
+      <LocalPackageConfigLine Include="&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;" />
+      <LocalPackageConfigLine Include="&lt;packages&gt;" />
+      <LocalPackageConfigLine Include="&lt;package id=&quot;%(Packages.PackageName)&quot; version=&quot;%(Packages.PackageVersion)&quot; /&gt;"/>
+      <LocalPackageConfigLine Include="&lt;/packages&gt;" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <RestoreConfig>%(Packages.InstallLocation)\packages.%(Packages.PackageName).config</RestoreConfig>
+    </PropertyGroup>
+    
+    <WriteLinesToFile File="$(RestoreConfig)" Lines="@(LocalPackageConfigLine)" Overwrite="true" />
+    <Message Text="Generated $(RestoreConfig) for '%(Packages.PackageName).%(Packages.PackageVersion).nupkg'" />
+    
+    <PropertyGroup>
+      <PackageSources>@(Packages, ';')</PackageSources>
+    </PropertyGroup>
+    <Message Text="Generated to package source to be '$(PackageSources)'"/>
+
+    <!-- Remove existing directories -->
+    <RemoveDir Directories="%(Packages.InstallLocation)\%(Packages.PackageName).%(Packages.PackageVersion)" />
```

Comment:
We may want to make this more general at some point perhaps adding a * at the end. However lets stick with this one for now.